chore: add new message api filtering docs (#548)
knocklabs/docs: content/reference.mdx

  <Attribute
    name="status"
    type="string[] (optional)"
    description="One or more of `queued`, `sent`, `delivered`, `delivery_attempted`, `undelivered`, `bounced`, `not_sent`. Limits results to messages with the given delivery status(es)."
  />
  <Attribute
    name="engagement_status"
    type="string[] (optional)"
    description="One or more of `read`, `unread`, `seen`, `unseen`. Limits results to messages with the given engagement status(es)."
  />
  <Attribute
    name="channel_id"
    type="string (optional)"
    description="Limits the results to only items that belong to the channel"
  />
  <Attribute
    name="messge_ids[]"
    type="string[] (optional)"

>     description="Limits the results to only the message ids given (max 50). Note: when using this option, all other filters will be"

Reviewer: Looks like this is missing a part of the sentence, 'all other filters will be...'

  <Attribute
    name="messge_ids[]"
    type="string[] (optional)"

>     description="Limits the results to only the message ids given (max 50). Note: when using this option, messages returned will also match any other filters applied."

Reviewer: ```suggestion
    description="Limits the results to only the message ids given (max 50). Note: when using this option, the results will be subject to the any other filters applied."
```
Hmmm...the `also` and `match` here make this feel additive when it will really only ever be fewer results b/c of other filters. Suggested one option above, but here's another: `when using this option, any other filters on the query will still be applied and further limit the result set.`

Reviewer 2: Good call out! I personally think the suggested change helps to clarify this.

    type="string[] (optional)"
    description="One or more of `queued`, `sent`, `delivered`, `delivery_attempted`, `undelivered`, `bounced`, `not_sent`. Limits results to messages with the given delivery status(es)."
  />
  <Attribute
    name="engagement_status"
    type="string[] (optional)"
    description="One or more of `read`, `unread`, `seen`, `unseen`. Limits results to messages with the given engagement status(es)."
  />
  <Attribute
    name="channel_id"
    type="string (optional)"
    description="Limits the results to only items that belong to the channel"
  />
  <Attribute
    name="messge_ids[]"
    type="string[] (optional)"

>     description="Limits the results to only the message ids given (max 50). Note: when using this option, all other filters will be"

Reviewer: same as above

  <Attribute
    name="status"
    type="string[] (optional)"
    description="One or more of `queued`, `sent`, `delivered`, `delivery_attempted`, `undelivered`, `bounced`, `not_sent`. Limits results to messages with the given delivery status(es)."
  />
  <Attribute
    name="engagement_status"
    type="string[] (optional)"
    description="One or more of `read`, `unread`, `seen`, `unseen`. Limits results to messages with the given engagement status(es)."
  />
  <Attribute
    name="channel_id"
    type="string (optional)"
    description="Limits the results to only items that belong to the channel"
  />
  <Attribute
    name="messge_ids[]"
    type="string[] (optional)"

>     description="Limits the results to only the message ids given (max 50). Note: when using this option, all other filters will be"

Reviewer: same as above
